kart/tabular/export.py

```diff
@@ -182,71 +203,88 @@ def table_export(
+    dataset_creation_options = list(dataset_creation_options)
+    layer_creation_options = list(layer_creation_options)
+    overwrite = _validate_overwrite(overwrite, layer_creation_options)
+    layer_name = layer_name or ds_path
 
     schema = dataset.schema
     pk_name = schema.first_pk_column.name
     primary_key_as_fid = _validate_primary_key_as_fid(primary_key_as_fid, schema)
 
     geom_key = None
     ogr_geom_type = None
-    ogr_crs = None
+    kart_crs = None
+    geometry_transform = None
 
-    if schema.has_geometry:
+    # Plan how to export geometry.
+    if schema.has_geometry and _should_export_geometry(driver, layer_creation_options):
         geom_col = schema.geometry_columns[0]
         geom_key = geom_col.name
         kart_geom_type = override_geometry_type or geom_col["geometryType"]
         ogr_geom_type = KART_GEOM_TYPE_TO_OGR_GEOM_TYPE[kart_geom_type.upper()]
-
-        crs_defs = list(dataset.crs_definitions().values())
-        if len(crs_defs) > 1:
-            raise CrsError(
-                f"Sorry, multiple CRS definitions at {ds_path!r} are not yet supported for export"
-            )
-        if crs_defs:
-            ogr_crs = make_crs(crs_defs[0], context=ds_path)
-
-        out_layer = out_ds.CreateLayer(
-            layer_name, ogr_crs, ogr_geom_type, options=list(layer_creation_option)
-        )
-    else:
-        out_layer = out_ds.CreateLayer(layer_name, options=list(layer_creation_option))
-
-    regular_keys = []
+        kart_crs = _get_dataset_crs(dataset)
+        if output_crs is None:
+            output_crs = kart_crs
+        else:
+            geometry_transform = _make_crs_transform(kart_crs, output_crs, ds_path)
+
+    # Plan how to export "regular" fields (not FID or geometry).
+    ogr_field_defns = {}
     for col in dataset.schema:
         if col.name == pk_name and not primary_key_as_field:
             continue
         if col.data_type == "geometry":
             continue  # Handled separately
-        regular_keys.append(col.name)
-        out_layer.CreateField(kart_schema_col_to_ogr_field_definition(col))
-
-    for feature in dataset.features_with_crs_ids(
-        repo.spatial_filter, show_progress=True
-    ):
-        out_feature = ogr.Feature(out_layer.GetLayerDefn())
-        if primary_key_as_fid:
-            out_feature.SetFID(feature[pk_name])
-        if geom_key:
-            geom = feature[geom_key]
-            if geom is None and drop_null_geometry_features:
-                continue
-            out_feature.SetGeometry(geom.to_ogr() if geom else None)
-        for i, key in enumerate(regular_keys):
-            out_feature.SetField(i, feature[key])
-        out_layer.CreateFeature(out_feature)
-
-    out_ds = None
+        ogr_field_defns[col.name] = kart_schema_col_to_ogr_field_definition(col)
+
+    # Finally, open the dataset for export.
+    try:
+        out_ds = open_dataset_for_export(
```

Comment:
> I'm unclear what this would do if the output already exists? I guess to allow multiple datasets to end up in the same FGDB/GPKG/etc then we should generally allow this.
For some formats the output is a directory (shapefiles); for others it's a file (GPKG), which complicates things a bit.
We don't want the working copy overwritten, that would be bad.

Logic around overwriting / updating datasets is now much improved = much more cautious.
See tests that make sure you can't overwrite an existing layer without trying to, and that you can add a layer to existing datasets where sensible (at least for the drivers tested. Drivers vary somewhat but hopefully we're testing a big enough range that the logic holds up okay).

As yet, no logic added to check if user is exporting to their working copy.
This is generally not what the user should be doing...
... but at least we prevent them overwriting the contents of their WC, unless they force it.
(Relatively tricky to detect in general if the destination is the working copy - would have to have special logic to check if two postgres:// strings resolve to the same place, for instance.)